### ISSUE TITLE
Duke Nukem Word Tour Invincible Enemy

### ISSUE THREAD

**Issue by Q3BFG10K** [ZDoom/Raze#696]
Win10-64, Raze 1.5.0, WT from Steam.

Load the save file (god mode is on) and fire as much as you like. The guy in front of you is totally invincible and never takes damage.

The first one took an extraordinary number of hits to kill, while this guy never dies.

I watched videos online and they both go down easily for others.

[Invicible.zip](https://github.com/ZDoom/Raze/files/9179874/Invicible.zip)

**Comment by coelckers** [comment]
We had this happen on occasion in the past but never were able to reliably reproduce it.
So right now I have no idea what goes wrong here on occasion.

**Comment by sinisterseed** [comment]
Indeed, this has been a complete nightmare since the beginning. The main issue is that we've not been able to find any way to reproduce it consistently, it sometimes is there, while others it is not, but what factors are involved in triggering this bug? We're just baffled. It also seems to happen with the World Tour version exclusively.

Did you do anything unusual, or maybe even nothing at all, something as seemingly harmless as continuing from a save on a different map, died/reloaded a few times, and so on? If we could finally reproduce it at will, we may be able to fix it at last...

**Comment by Q3BFG10K** [comment]
I can't think of any unusual actions that caused it.

Is it possible he's stuck inside geometry, so the game thinks you're shooting at a wall? If you move around and observe his movement, he never seems to move too far from that position.

I never tried this, but is he still invincible if you load the save in eDuke?

**Comment by sinisterseed** [comment]
The formats differ, loading a save file from Raze into another port will not work.

It could be a possibility, I recall seeing something to that effect in an old report where a Battlelord turned invincible and was also ignoring the player's presence, but one of the times I managed to trigger it by accident, the enemy could move freely and behaved normally, while being simply unkillable.

As I said, it's a bug that leaves everyone confused as we're yet to determine the conditions which trigger its appearance.

**Comment by coelckers** [comment]
The main problem with all this is that it's all being controlled from the CON scripts - if they somehow managed to get into a state where certain checks don't get performed anymore, an actor will become broken.

**Comment by Gaerzi** [comment]
It's probably a problem in the CON script logic, like the one in Nuclear Winter where the Flying Frosty is immune to ice death but not immune to ice damage, so if he gets killed by ice damage (including from his own shots ricocheting at him...) he just becomes invulnerable. Being already dead, you can't kill him, but he still behaves like he's alive.

If this is a problem of this kind, then there's no easy fix, short of having a system to override CON code and include a fixed version... Which also means first fixing the CON code.

**Comment by sinisterseed** [comment]
That may, or may not be the case. The idea is that, at least to my knowledge, there never have been any reports of this issue happening with the existing ports and the original World Tour, which makes me believe it's something we may have broken somewhere all the way back when we originally added support for it, somehow.

I've seen it with freshly spawned enemies not touched by anyone or anything but being simply invulnerable, which makes me suspect this is different from the bug in Nuclear Winter.
